src/magic_spell_effect.cpp

```diff
@@ -112,6 +112,15 @@ void spell_effect::teleport_random( const spell &sp, Creature &caster, const tri
     teleport::teleport( caster, min_distance, max_distance, safe, false );
 }
 
+void spell_effect::swap_pos( const spell &, Creature &caster, const tripoint &target )
```

Comment:
if you're not using this as a spell_effect, the unused spell parameter isn't necessary to define. additionally... i think it can be static since it's not used anywhere else.